service/common/src/main/java/org/apache/polaris/service/events/AwsCloudWatchEventListener.java

```diff
@@ -0,0 +1,264 @@
+import software.amazon.awssdk.services.cloudwatchlogs.model.UnrecognizedClientException;
+
+@ApplicationScoped
+@Identifier("aws-cloudwatch")
+public class AwsCloudWatchEventListener extends PolarisEventListener {
+  private static final Logger LOGGER = LoggerFactory.getLogger(AwsCloudWatchEventListener.class);
+  private final ObjectMapper objectMapper = new ObjectMapper();
+  private static final int MAX_BATCH_SIZE = 10_000;
+  private static final int MAX_WAIT_MS = 5000;
+  private static final String DEFAULT_LOG_STREAM_NAME = "polaris-cloudwatch-default-stream";
+  private static final String DEFAULT_LOG_GROUP_NAME = "polaris-cloudwatch-default-group";
+  private static final String DEFAULT_REGION = "us-east-1";
+
+  private final BlockingQueue<EventWithTimestamp> queue = new LinkedBlockingQueue<>();
+  private CloudWatchLogsClient client;
+  private volatile String sequenceToken;
+
+  private volatile boolean running = true;
+
+  ExecutorService executorService;
+
+  private Future<?> backgroundTask;
+
+  private final String logGroup;
+  private final String logStream;
+  private final Region region;
+
+  @Inject
+  public AwsCloudWatchEventListener(
+      EventListenerConfiguration config, ExecutorService executorService) {
+    this.executorService = executorService;
+
+    this.logStream = config.awsCloudwatchlogStream().orElse(DEFAULT_LOG_STREAM_NAME);
+    this.logGroup = config.awsCloudwatchlogGroup().orElse(DEFAULT_LOG_GROUP_NAME);
+    this.region = Region.of(config.awsCloudwatchRegion().orElse(DEFAULT_REGION));
+  }
+
+  @PostConstruct
+  void start() {
+    this.client = createCloudWatchClient();
+    ensureLogGroupAndStream();
+    backgroundTask = executorService.submit(this::processQueue);
+  }
+
+  protected CloudWatchLogsClient createCloudWatchClient() {
+    return CloudWatchLogsClient.builder().region(region).build();
+  }
+
+  private void processQueue() {
+    List<EventWithTimestamp> drainedEvents = new ArrayList<>();
+    List<InputLogEvent> transformedEvents = new ArrayList<>();
+    while (running || !queue.isEmpty()) {
+      drainQueue(drainedEvents, transformedEvents);
+    }
+  }
+
+  @VisibleForTesting
+  public void drainQueue(
+      List<EventWithTimestamp> drainedEvents, List<InputLogEvent> transformedEvents) {
+    try {
+      EventWithTimestamp first = queue.poll(MAX_WAIT_MS, TimeUnit.MILLISECONDS);
+
+      if (first != null) {
+        drainedEvents.add(first);
+        queue.drainTo(drainedEvents, MAX_BATCH_SIZE - 1);
+      } else {
+        return;
+      }
+
+      drainedEvents.forEach(event -> transformedEvents.add(createLogEvent(event)));
+
+      sendToCloudWatch(transformedEvents);
+    } catch (InterruptedException e) {
+      if (!queue.isEmpty()) {
+        LOGGER.debug("Interrupted while waiting for queue to drain", e);
+        queue.addAll(drainedEvents);
+      }
+    } catch (DataAlreadyAcceptedException e) {
+      LOGGER.debug("Data already accepted: {}", e.getMessage());
```

Comment:
In this case, isn't it possible only _some_ of the events were already accepted?